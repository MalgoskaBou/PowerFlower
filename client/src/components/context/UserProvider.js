import React, { createContext, useState, useEffect } from "react";
import { useQuery } from "@apollo/react-hooks";
import currentUserQuery from "../../queries/currentUser";
const context = createContext(null);

const UserProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const { loading, data } = useQuery(currentUserQuery);

  useEffect(() => {
    setUser({ user: data?.currentUser, loading });
  }, [loading, data]);

  return <context.Provider value={user}>{children}</context.Provider>;
};

UserProvider.context = context;

export default UserProvider;
