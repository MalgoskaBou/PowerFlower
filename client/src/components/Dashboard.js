import React, { useContext } from "react";
import UserProvider from "./context/UserProvider";

const Dashboard = () => {
  const userData = useContext(UserProvider.context);

  return (
    <div>
      Dashboard - Your zones:
      {userData.user.zones?.map(({ name, id }) => (
        <p key={id}>{name}</p>
      ))}
      <p>{`${userData.user.confirmed}`}</p>
      <p>{`${userData.user.name}`}</p>
    </div>
  );
};

export default Dashboard;
