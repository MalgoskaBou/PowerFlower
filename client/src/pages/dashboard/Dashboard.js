import React, { useContext } from "react";
import { useMutation } from "@apollo/react-hooks";
import UserProvider from "../../components/context/UserProvider";
import logoutUserMutation from "../../queries/logoutUser";
import Wrapper from "../../components/general-style/Wrapper";
import IconButton from "../../components/general-style/IconButton";
import FlexWrapper from "../../components/general-style/FlexWrapper";
import Button from "../../components/general-style/Button";
import Zone from "../../components/general-style/Zone";

const Dashboard = () => {
  const userData = useContext(UserProvider.context);
  const [logoutUser] = useMutation(logoutUserMutation);

  const handleLogoutUser = async () => {
    try {
      await logoutUser();
      window.location.reload();
    } catch (err) {
      console.log(err);
    }
  };

  return (
    <Wrapper>
      <div>
        <Button fontSize={1.4} smallerLetterSpace>
          Dodaj przestrze≈Ñ
        </Button>
        <IconButton icon="icon-user" onClick={handleLogoutUser}>
          Wyloguj
        </IconButton>
        <FlexWrapper>
          {userData.user.zones?.map(({ name, id }) => (
            <Zone key={id} name={name} />
          ))}
        </FlexWrapper>
        <p>{`${userData.user.confirmed}`}</p>
        <p>{`${userData.user.name}`}</p>
      </div>
    </Wrapper>
  );
};

export default Dashboard;
