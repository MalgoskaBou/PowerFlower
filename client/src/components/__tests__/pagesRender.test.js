import React from "react";
import { render, waitForElement } from "@testing-library/react";
import { MockedProvider } from "@apollo/react-testing";
import { BrowserRouter as Router } from "react-router-dom";
import LoginScreen from "../../pages/loginScreen/LoginForm";
import Dashboard from "../../pages/dashboard/Dashboard";
import loginUserMutation from "../../queries/loginUser";
import currentUserQuery from "../../queries/currentUser";
import UserProvider from "../context/UserProvider";
import renderer from "react-test-renderer";

const loginUserMutationMock = [
  {
    request: {
      query: loginUserMutation,
    },
  },
];

const currentUserQueryMock = [
  {
    request: {
      query: currentUserQuery,
    },
    result: {
      data: {
        currentUser: {
          name: "kitek",
          confirmed: true,
          zones: [
            {
              id: "5e275b818ad8e80a0079a7c6",
              name: "Zone1",
            },
            {
              id: "5e5bf5ec40b37f43b4947dd6",
              name: "Zone2",
            },
          ],
        },
      },
    },
  },
];

async function wait(ms = 0) {
  await renderer.act(() => {
    return new Promise((resolve) => {
      setTimeout(resolve, ms);
    });
  });
}

describe("Pages render correctly", () => {
  it("Login screen render correctly", () => {
    const { getByPlaceholderText, getByText } = render(
      <MockedProvider mocks={loginUserMutationMock} addTypename={false}>
        <LoginScreen />
      </MockedProvider>
    );
    expect(getByPlaceholderText(/e-mail/i)).toBeInTheDocument();
    expect(getByPlaceholderText(/Password/i)).toBeInTheDocument();
    expect(getByText(/Log in/i)).toBeInTheDocument();
    expect(getByText(/flowerLogin.svg/i)).toBeInTheDocument();
  });

  it("Dashboard screen render correctly", async () => {
    const { getByText } = render(
      <Router>
        <MockedProvider mocks={currentUserQueryMock} addTypename={false}>
          <UserProvider>
            <Dashboard />
          </UserProvider>
        </MockedProvider>
      </Router>
    );

    const zone = await waitForElement(() => getByText(/zone1/i));
    expect(zone).toBeInTheDocument();
    expect(getByText(/wyloguj/i)).toBeInTheDocument();
  });

  it("Dashboard snapshot", async () => {
    const tree = renderer
      .create(
        <Router>
          <MockedProvider mocks={currentUserQueryMock} addTypename={false}>
            <UserProvider>
              <Dashboard />
            </UserProvider>
          </MockedProvider>
        </Router>
      )
      .toJSON();

    await wait();

    expect(tree).toMatchSnapshot();
  });

  it("LoginScreen snapshot", async () => {
    const tree = renderer
      .create(
        <MockedProvider mocks={currentUserQueryMock} addTypename={false}>
          <LoginScreen />
        </MockedProvider>
      )
      .toJSON();

    await wait();

    expect(tree).toMatchSnapshot();
  });
});
