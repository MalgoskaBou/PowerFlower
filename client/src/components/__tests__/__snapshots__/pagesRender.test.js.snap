// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages render correctly Dashboard snapshot 1`] = `
<section
  className="sc-AykKF gyCxNz"
>
  <div
    className="sc-fzXfLT kwKusS"
  >
    <button
      className="sc-AykKD ciZAXx"
      fontSize={1.4}
    >
      Dodaj przestrze≈Ñ
    </button>
    <button
      className="sc-AykKH eyrVSJ"
      onClick={[Function]}
    >
      Wyloguj - 
      undefined
      <i
        className="icon-user icon"
      />
    </button>
  </div>
  <section
    className="sc-AykKI jmDIge"
  />
</section>
`;

exports[`Pages render correctly LoginScreen snapshot 1`] = `
<section
  className="sc-AykKF sc-AykKG gTdQJv"
>
  <svg>
    flowerLogin.svg
  </svg>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <input
      className="sc-AykKE bVzCnM"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="e-mail"
      type="text"
      value=""
    />
    <input
      className="sc-AykKE fESoYg"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <button
      className="sc-AykKD dwveSc"
      type="submit"
      value="Log in"
    >
      Log in
    </button>
  </form>
  <a
    className="sc-AykKC fJjnQA"
  >
    Forgot password
  </a>
  <span
    style={
      Object {
        "fontSize": "2.4rem",
        "height": 0,
        "zIndex": -9999,
      }
    }
  >
    
  </span>
</section>
`;
